src/instrumenter/interpose.ts

```diff
@@ -82,16 +82,34 @@ function callOriginal(
     return factory.makeExpressionStatement(assignment);
 }
 
-function renameReturns(factory: ASTNodeFactory, stub: FunctionDefinition): Recipe {
-    return stub.vReturnParameters.vParameters
-        .filter((ret) => ret.name === "")
-        .map((_, idx) => new RenameReturn(factory, stub, idx, `RET_${idx}`));
+function renameReturns(
+    factory: ASTNodeFactory,
+    stub: FunctionDefinition,
+    varsInScope: Set<string>
+): Recipe {
+    const returnVars = stub.vReturnParameters.vParameters.filter((ret) => ret.name === "");
+    let varId = 0;
+    const renamedVars = [];
+    for (let i = 0; i < returnVars.length; i++) {
+        while (varsInScope.has(`RET_${varId}`)) {
```

Comment:
Instead of a while loop here perhaps we can use `getNewVarName(varsInScope, `RET`)`